ocpp1.6/charge_point.go

```diff
@@ -334,7 +334,7 @@ func (cp *chargePoint) sendResponse(confirmation ocpp.Response, err error, reque
 
 func (cp *chargePoint) Start(centralSystemUrl string) error {
 	// Overriding some protocol-specific values in the lower layers globally
-	ocppj.FormationViolation = ocppj.FormatViolationV16
+	cp.client.Endpoint.SetDialect(ocpp.V16)
```

Comment:
I would honestly move this logic to the `ocpp16.NewChargePoint` constructor function (respectively for both endpoints for both proto versions). It's a static setting that should be set at creation time, along with the sub-protocol and base handlers.

Reply:
Brilliant. Thought about that but could not finde the right place to do so.